GitCommands/Submodules/SubmoduleStatusProvider.cs

```diff
@@ -190,63 +191,70 @@ private async Task GetSubmoduleStatusAsync(bool updateStatus, SubmoduleInfo info
 
         private async Task GetSuperProjectRepositorySubmodulesStatusAsync(bool updateStatus, SubmoduleInfoResult result, GitModule module, CancellationToken cancelToken, string noBranchText)
         {
-            if (module.SuperprojectModule == null)
+            // This function always at least sets result.TopProject
+
+            bool isCurrentTopProject = module.SuperprojectModule == null;
+            if (isCurrentTopProject)
             {
+                string path = module.WorkingDir;
+                string name = Path.GetFileName(Path.GetDirectoryName(module.WorkingDir)) + GetBranchNameSuffix(path);
+                result.TopProject = new SubmoduleInfo { Text = name, Path = module.WorkingDir, Bold = true };
                 return;
             }
 
-            string name, path;
-            GitModule supersuperproject = FindTopProjectModule(module.SuperprojectModule);
-            if (module.SuperprojectModule.WorkingDir != supersuperproject.WorkingDir)
+            IGitModule topProject = FindTopProjectModule(module.SuperprojectModule);
+            bool isParentTopProject = module.SuperprojectModule.WorkingDir == topProject.WorkingDir;
+
+            // Set result.SuperProject
```

Comment:
This isn't something we do in the codebase and this may give other contributors some wrong ideas.
Any reason we can't extract the functionality into (not inlined) methods?